### PR TITLE
Added link check to CI

### PR DESCRIPTION
**Pull request by PurpleBriar**
## Type of PR

- [x] Documentation changes
- [ ] Code changes
- [ ] Test changes
- [x] CI-CD changes
- [ ] GitHub Template changes

### Purpose of PR
Added link check to CI. Updated dead links in markdown files

### Validation
- [ ] Unit tests updated and ran successfully
- [x] Update documentation or issue referenced above

### Issues Closed or Referenced

- References #helium/issues/542

### CODE REVIEW

File: mlc_config.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "ignorePatterns": [
+        {
+            "pattern": "https://github.com/\\["
+        }
```
Comment (aflinchb) [inline mlc_config.json:5]: What is this file used for?

**Review by aflinchb — APPROVED** [review]
Approved w/ one question

File: mlc_config.json (hunk 1 above)
Comment (PurpleBriar) [reply mlc_config.json:5]: This is used to define patterns for the link checker to ignore. In this case, I wanted it to ignore paths like _**https://github.com/[variable]**_, which would be incorrectly flagged as dead links.

Comment (aflinchb) [reply mlc_config.json:5]: Got it - thanks!

**Review by sivamu — APPROVED** [review]

**Pull request merged**